src/main/java/org/folio/dataexp/service/QuickExportService.java

```diff
@@ -0,0 +1,80 @@
+import org.folio.dataexp.domain.dto.QuickExportResponse;
+import org.folio.dataexp.domain.entity.ExportIdEntity;
+import org.folio.dataexp.exception.export.DataExportRequestValidationException;
+import org.folio.dataexp.repository.ExportIdEntityRepository;
+import org.folio.dataexp.repository.JobExecutionEntityRepository;
+import org.folio.dataexp.repository.JobProfileEntityRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+import static java.util.Objects.nonNull;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class QuickExportService {
+
+  private final FileDefinitionsService fileDefinitionsService;
+  private final DataExportService dataExportService;
+  private final JobProfileEntityRepository jobProfileEntityRepository;
+  private final ExportIdEntityRepository exportIdEntityRepository;
+  private final JobExecutionEntityRepository jobExecutionEntityRepository;
+
+  public QuickExportResponse postQuickExport(QuickExportRequest quickExportRequest) {
+    var fileDefinition = new FileDefinition().id(UUID.randomUUID()).size(0).fileName(quickExportRequest.getType() + "-quick.csv");
+    fileDefinitionsService.postFileDefinition(fileDefinition);
+    log.info("Post quick export for job profile {}", quickExportRequest.getJobProfileId());
+    dataExportService.postDataExport(getExportRequestFromQuickExportRequest(quickExportRequest, fileDefinition));
+    var jobExecution = jobExecutionEntityRepository.getReferenceById(fileDefinition.getJobExecutionId());
+    return QuickExportResponse.builder().jobExecutionId(fileDefinition.getJobExecutionId())
+      .jobExecutionHrId(jobExecution.getJobExecution().getHrId()).build();
+  }
+
+  private ExportRequest getExportRequestFromQuickExportRequest(QuickExportRequest quickExportRequest, FileDefinition fileDefinition) {
+    var exportRequest = new ExportRequest();
+    saveBatch(quickExportRequest, fileDefinition);
+    exportRequest.setJobProfileId(getDefaultJobProfileId(quickExportRequest));
+    exportRequest.setRecordType(ExportRequest.RecordTypeEnum.fromValue(quickExportRequest.getRecordType().getValue()));
+    exportRequest.setFileDefinitionId(fileDefinition.getId());
+    exportRequest.setQuick(true);
+    exportRequest.setIdType(ExportRequest.IdTypeEnum.fromValue(quickExportRequest.getRecordType().getValue().toLowerCase()));
+    return exportRequest;
+  }
+
+  private void saveBatch(QuickExportRequest quickExportRequest, FileDefinition fileDefinition) {
+    var uuids = quickExportRequest.getUuids();
+    if (nonNull(uuids)) {
+      var batch = new ArrayList<ExportIdEntity>();
+      uuids.forEach(instanceId -> {
+        var entity = ExportIdEntity.builder().jobExecutionId(fileDefinition
+          .getJobExecutionId()).instanceId(instanceId).build();
+        batch.add(entity);
+      });
+      exportIdEntityRepository.saveAll(batch);
+    } else {
+      log.error("Nothing to export: no uuids provided.");
```

Comment:
Please, add file definition id to the error message.

Reply:
Done.